testing/src/main/java/org/oppia/android/testing/logging/EventLogSubject.kt

```diff
@@ -1944,6 +1965,64 @@ class EventLogSubject private constructor(
     }
   }
 
+  /**
+   * Truth subject for verifying properties of [EventLog.FeatureFlagContext]s.
+   *
+   * Note that this class is also a [LiteProtoSubject] so other aspects of the underlying
+   * [EventLog.FeatureFlagContext] proto can be verified through inherited methods.
+   *
+   * Call [FeatureFlagContextSubject.assertThat] to create the subject.
+   */
+  class FeatureFlagContextSubject private constructor(
+    metadata: FailureMetadata,
+    private val actual: EventLog.FeatureFlagContext
+  ) : LiteProtoSubject(metadata, actual) {
+    /**
+     * Returns a [StringSubject] to test [EventLog.FeatureFlagContext.getSessionId].
+     *
+     * This method never fails since the underlying property defaults to empty string if it's not
+     * defined in the context.
+     */
+    fun hasSessionIdThat(): StringSubject = assertThat(actual.sessionId)
+
+    /**
+     * Returns a [Subject] to test the  of the properties of a feature flag with the passed
+     * name in the [EventLog.FeatureFlagContext] list.
+     */
+    fun hasNamedFeatureFlagThat(name: String): Subject {
```

Comment:
It  goes against the patterns that we have to have logic in this class. The actual name of the flag to be verified should be passed in the test, and it would read something like:

```kotlin
  EventLogSubject.assertThat(event).FeatureFlagContext {
        hasSessionIdThat().isEqualTo(TEST_SESSION_ID)
        hasFeatureFlagWithNameThat().isEqualTo(DOWNLOADS_SUPPORT)
        ...
      }
```

If we wanted to check that ALL the feature flag names were logged, we could add the assertions for each in the list.

Reply:
I'm not sure if I implemented this the way you suggested. PTAL and let me know if it is okay now.